### ISSUE TITLE
Add Type Hints

### ISSUE THREAD

**Issue by bionicles** [rdkit/rdkit#7682]
**Is your feature request related to a problem? Please describe.**
It's hard to know the types of the inputs and outputs to various functions

**Describe the solution you'd like**
- [ ] Add type hints to all  `rdkit` functions
- [ ] Format all `rdkit` functions with Black or Flake8 formatter (ruff is fast)

**Describe alternatives you've considered**
Someone mentioned using jedi

**Additional context**
[Add any other context or screenshots about the feature request here.
](https://github.com/rdkit/rdkit/discussions/3981#discussion-3293816)

**Comment by greglandrum** [comment]
There are already stubs available. They are described here:
https://greglandrum.github.io/rdkit-blog/posts/2024-03-08-docstring-stubs.html

We use yapf to format the code, which is also a "standard format"

**Comment by bionicles** [comment]
you're right! weird, i must have used "jump to definition" on a rare case where there's no stub (Draw.MolToImage) and it jumped here and I saw this return type annotation was super complicated for some reason

![image](https://github.com/user-attachments/assets/896c4a8f-1236-47f1-a74f-021bdd3d5bd7)

found this because i was confused about the PIL.Image module containing an identically named class Image (whoever called it PIL.Image.Image is a naming king) in this context:
![image](https://github.com/user-attachments/assets/8dcbc3b1-5b77-48ee-ad43-27e0ad56a0cb)

Could it be worthwhile to merge type hints into the source instead of using stubs, it's annoying to jump to a definition and just get a stub, makes it harder to tinker with the code?

Likely well past the point where it's possible to rename things majorly, and would be a breaking change, but what if we refactor names for functions to use snake_case instead of PascalCase to avoid confusing functions with classes / types?

**Comment by greglandrum** [comment]
We will not be changing the naming scheme or adding type annotations to the source.

**Comment by bionicles** [comment]
alright, fair enough, no problemo, i'll rename the issue in a more minimal way

there's a stub for MolToImage on line 82 of `rdkit-stubs/Chem/Draw/__init__.pyi`
but it seems to not be working
